test/yaml-mode-test.el

```diff
@@ -0,0 +1,117 @@
+;;;; yaml-mode-test.el --- Tests for yaml-mode  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2010-2022 Yoshiki Kurihara
+
+;; Author: Yoshiki Kurihara <clouder@gmail.com>
+;;         Marshall T. Vandegrift <llasram@gmail.com>
+;; Maintainer: Vasilij Schneidermann <mail@vasilij.de>
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; many test utilities are copied from markdown-mode
+
+;;; Code:
+
+(require 'yaml-mode)
+(require 'ert)
+(require 'cl-lib)
+
+;; for version < 25
+(defconst yaml-test-font-lock-function
+  (if (fboundp 'font-lock-ensure)
+      #'font-lock-ensure #'font-lock-fontify-buffer))
+
+(defmacro yaml-test-string-mode (mode string &rest body)
```

Comment:
This is a premature abstraction. Since only yaml-mode is tested, both macros can be combined into one:

- Change the `(funcall ,mode)` line to `(yaml-mode)`
- Rename this macro to `yaml-test-string`
- Delete the other macro